### PR TITLE
Bump IBC, Cosmos-SDK, Tendermint to latest versions

### PR DESCRIPTION
**Pull request by faddat**
Since IBC v1.0.2, there have been multiple security rleases of IBC-Go.  This bumps that to 1.2.2 and bumps Tendermint and the Cosmos SDK to 0.34.14 and 0.44.3, respectively.  Together, these will make for a more performant node, especially from the perspective of a relayer.﻿

### CODE REVIEW

**Review by askolesov — APPROVED** [review]

**Comment by askolesov** [comment]
Thanks! @ankurdotb Are you OK with merging this?

File: go.mod [hunk 1]
```diff
@@ -17,12 +16,113 @@ require (
...
+	golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a // indirect
+	golang.org/x/net v0.0.0-20210903162142-ad29c8ab022f // indirect
+	golang.org/x/sys v0.0.0-20210903071746-97244b99971b // indirect
+	golang.org/x/term v0.0.0-20201126162022-7de9c90e9dd1 // indirect
+	golang.org/x/text v0.3.6 // indirect
+	google.golang.org/protobuf v1.27.1 // indirect
+	gopkg.in/ini.v1 v1.62.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+	nhooyr.io/websocket v1.8.6 // indirect
```
Comment (ankurdotb) [inline go.mod:123]: This library is the one that is dependent on vulnerable version of `gin-gonic` but doesn't seem to be used any more in Cosmos SDK or Tendermint upstream.

Cosmos fix in v0.44.3: https://github.com/cosmos/cosmos-sdk/blob/b75c29fc15d3320ec0c7596dbd7c787c48dccad8/go.mod#L67

Tendermint also doesn't use it.

**Review by ankurdotb — CHANGES_REQUESTED** [review]
@Toktar @askolesov @anikitinDSR Also cross-check the dependencies for go.mod, remove any unnecessary and if there's any other versions bumped to higher upstream, change those as well.

**Comment by ankurdotb** [comment]
This might be superseded by https://github.com/cheqd/cheqd-node/pull/186 if everything is incorporporated there.

**Comment by ankurdotb** [comment]
@faddat The go version suggested is also higher than that used by Cosmos SDK.

**Comment by ankurdotb** [comment]
Resolved under https://github.com/cheqd/cheqd-node/pull/186 and superseded

**Pull request closed without merging**